### PR TITLE
Add InferRequest tests, setters and getters

### PR DESCRIPTION
**Pull request by almilosz**
### Details:
 - Expose `setTensor()`, `setInputTensor()`, `setOutputTensor()`, `get_input_tensor()`, `getCompiledModel`
 - Add Node.js tests for setters and getters
 - Move InferRequest tests to `infer_request.test.js`
 - Move helper method to `utils.js`

### Tickets:
 - [117428]

### CODE REVIEW

File: src/bindings/js/node/tests/infer_request.test.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+const ov = require('../build/Release/ov_node_addon.node');
+const assert = require('assert');
+const { describe, it } = require('node:test');
+const { getModelPath } = require('./utils.js');
+
+describe('InferRequest', () => {
+
+  let testXml = getModelPath();
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/infer_request.test.js:12]: Maybe I forgot why here cannot use const, just ignore if cannot
```suggestion
  const testXml = getModelPath();
```

File: src/bindings/js/node/tests/infer_request.test.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  const inferRequest = compiledModel.createInferRequest();
+
+  const tensorData = Float32Array.from({ length: 3072 }, () => Math.floor(Math.random() * 3072));
+  const tensor = new ov.Tensor(
+    ov.element.f32,
+    Int32Array.from([1, 3, 32, 32]),
+    tensorData,
+  );
+  const res_tensor = new ov.Tensor(
```
Comment (vishniakov-nikolai) [inline src/bindings/js/node/tests/infer_request.test.js:25]:
```suggestion
  const resTensor = new ov.Tensor(
```

**Review by vishniakov-nikolai — APPROVED** [review]

File: src/bindings/js/node/tests/infer_request.test.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    assert.deepStrictEqual(tensor.data[0], t1.data[0]);
+    assert.deepStrictEqual(tensor.data[0], t2.data[0]);
+    assert.deepStrictEqual(tensor.data[0], t3.data[0]);
+
+    ir.infer();
+    const res1 = ir.getOutputTensor();
+    const res2 = ir.getOutputTensor(0);
+    assert.deepStrictEqual(res1.data[0], res2.data[0]);
+  });
```
Comment (jiwaszki) [inline src/bindings/js/node/tests/infer_request.test.js:101]: Make them separate.

File: src/bindings/js/node/lib/addon.ts [hunk 4]
```diff
@@ -44,15 +44,19 @@ interface TensorConstructor {
...
 
-
 interface InferRequest {
-  // FIXME: are we going to add index parameter for this method?
-  getOutputTensor(): Tensor;
-  getOutputTensors(): Tensor[];
-  getTensor(output: Output): Tensor;
+  setTensor(name: string, tensor: Tensor): void;
+  setInputTensor(idxOrTensor: number | Tensor, tensor?: Tensor): void;
+  setOutputTensor(idxOrTensor: number | Tensor, tensor?: Tensor): void;
```
[the remark this answers is not in the document]
Comment (almilosz) [reply src/bindings/js/node/lib/addon.ts:50]: Added

**Pull request merged**